### PR TITLE
disable proxy for k8s only when configured.

### PR DESCRIPTION
**Pull request by huddy**
This PR improves the proxy support in #73 to give some flexibility in when the proxy is used. In our case we have the following requirements:

1. Calls to AWS APIs must use `HTTP_PROXY` since we have no direct internet access in our VPCs 
2. Calls to k8s cluster must not go via `HTTP_PROXY`. 

In this case we must have `HTTP_PROXY` set but this causes the rolling update code to try and also go via `HTTP_PROXY`. `NO_PROXY` is configured to include the k8s cluster endpoint but isn't adhered to here. 

I wanted to adhere to `NO_PROXY` originally but I couldn't see a good way get `servers` value from kubernetes python client once it loaded config, to see if it matched something in `NO_PROXT`. I also didn't think it would be a good idea read kube config independently. 

For the above reasons I went with an additional environment variable `N0_K8S_PROXY_PASS`. 

I'm happy to change tactics here if you think there's a better way.

### CODE REVIEW

File: eksrollup/lib/k8s.py [hunk 1]
```diff
@@ -18,7 +18,9 @@ def ensure_config_loaded():
             raise Exception("Could not configure kubernetes python client")
 
     proxy_url = os.getenv('HTTPS_PROXY', os.getenv('HTTP_PROXY', None))
-    if proxy_url:
+    no_k8s_proxy_pass = os.getenv('N0_K8S_PROXY_PASS')
```
Comment (crhuber) [inline eksrollup/lib/k8s.py:21]:
```suggestion
    k8s_proxy_bypass = os.getenv('K8S_PROXY_BYPASS')
```

File: eksrollup/lib/k8s.py [hunk 2]
```diff
@@ -18,7 +18,9 @@ def ensure_config_loaded():
...
+
+    if proxy_url and no_k8s_proxy_pass != "true":
```
Comment (crhuber) [inline eksrollup/lib/k8s.py:23]:
```suggestion
    if proxy_url and k8s_proxy_bypass != "true":
```

File: tests/test_k8s.py [hunk 3]
```diff
@@ -35,6 +35,12 @@ def test_ensure_config_loaded_proxy_fallback_to_http(self):
             ensure_config_loaded()
             self.assertEqual('http://localhost:12345', ApiClient().configuration.proxy)
 
+    def test_ensure_config_loaded_proxy_not_set_when_no_k8s_set(self):
+        with patch.dict(os.environ, {'N0_K8S_PROXY_PASS': 'true', 'HTTP_PROXY': 'http://localhost:6789'}):
```
Comment (crhuber) [inline tests/test_k8s.py:39]:
```suggestion
        with patch.dict(os.environ, {'K8S_PROXY_BYPASS': 'true', 'HTTP_PROXY': 'http://localhost:6789'}):
```

**Comment by crhuber** [comment]
hi @huddy thanks for the contribution. If you can also add to the documentation about what K8S_PROXY_BYPASS does that would be great. 
I wasn't able to test is because we dont fit this use case, but @chadlwilson do you have any issues with this?

File: eksrollup/lib/k8s.py (hunk 2 above)
Comment (huddy) [reply eksrollup/lib/k8s.py]: Foxed

File: tests/test_k8s.py (hunk 3 above)
Comment (huddy) [reply tests/test_k8s.py]: Updated :)

**Comment by huddy** [comment]
> hi @huddy thanks for the contribution. If you can also add to the documentation about what K8S_PROXY_BYPASS does that would be great.
> I wasn't able to test is because we dont fit this use case, but @chadlwilson do you have any issues with this?

Hey @crhuber thanks for the review! Really appreciate that. 

I've updated `README.md` and made the changes you requested.

**Pull request merged**